src/shared/generator.c

```diff
@@ -708,6 +709,85 @@ int generator_hook_up_pcrfs(
         return generator_add_symlink_full(dir, where_unit, "wants", pcrfs_unit_path, instance);
 }
 
+int generator_hook_up_quotacheck(
+                const char *dir,
+                const char *what,
+                const char *where,
+                const char *target,
+                const char *fstype) {
+
+        _cleanup_free_ char *where_unit = NULL, *instance = NULL;
+        int r;
+
+        assert(dir);
+        assert(where);
+
+        if (isempty(fstype) || streq(fstype, "auto"))
+                return log_warning_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Couldn't determine filesystem type for %s, quota cannot be activated", what);
+        if (!fstype_needs_quota(fstype))
+                return log_warning_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Quota was requested for %s, but not supported, ignoring: %s", what, fstype);
+
+        r = unit_name_from_path(where, ".mount", &where_unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to make unit name from path '%s': %m", where);
+
+        /* quotacheck unit for system root */
+        if (path_equal(where, "/"))
+                return generator_add_symlink(dir, where_unit, "wants", SPECIAL_QUOTACHECK_ROOT_SERVICE);
+
+        r = unit_name_path_escape(where, &instance);
+        if (r < 0)
+                return log_error_errno(r, "Failed to escape path '%s': %m", where);
+
+        r = generator_add_ordering(dir, where_unit, "After", SPECIAL_QUOTACHECK_SERVICE, instance);
+        if (r < 0)
+                return r;
```

Comment:
so hmm, so here you are sayint he mount unit shall be ordered after its own quotacheck service, and you use the mount path as argument to quotacheck. how is that going to work? for the path to be valid and accessible it means it must be mounted, hence mounting needs to take place *before* quotacheck, not after...

Reply:
and this ordering dep can be implemented in the unit file template anyway afaics, no need to generate it problematic, it's the same as just adding `Before=%i.mount` in the quotacheck unit file.

(but as mentioned, I think it's wrong either way)

Reply:
Sorry, then this was a mistake. The mount unit needs to be ordered before the quotacheck unit.
I will add it to the template then.